app/controllers/settings_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class SettingsController < ApplicationController
+  before_action :authenticate_user!
+  include OutsideDevise
+
+  def new
+    @addresses = AddressesForm.new(user_id: current_user.id)
+  end
+
+  def create
+    @addresses = AddressesForm.new(addresses_params)
+    status = @addresses.save ? :created : :unprocessable_entity
+    render json: @addresses.errors, callback: 'parse_errors', status: status
+  end
+
+  private
+
+  def addresses_params
+    %i[shipping billing].each { |type| params[type][:user_id] = current_user.id }
```

Comment:
it's should be in form object

Reply:
I wrote the [AddressesForm](https://github.com/GalenkoNeon/book_store/blob/dev/app/forms/addresses_form.rb)(form object) that it can store addresses both for the user in the `settings` and in the `checkout` for the particular Order, all the logic is based on accepted parameters. If I  pass user_id, it will save addresses for user(MyAccount/Settings), if order_id it will save as order addresses(Checkout/AddressesTab), If nothing is passed on, an empty object for the form will be initialized `AddressesForm.new()`
Do I need make it different?